mzLib/Omics/IBioPolymerWithSetMods.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Text;
+using Chemistry;
+using MassSpectrometry;
+using Omics.Digestion;
+using Omics.Fragmentation;
+using Omics.Modifications;
+
+namespace Omics
+{
+    /// <summary>
+    /// Interface for modified and unmodified precursor ions
+    /// </summary>
+    /// <remarks>
+    /// Proteins -> PeptideWithSetModifications : ProteolyticPeptide
+    /// Nucleic Acids -> OligoWithSetMods : NucleolyticOligo
+    /// </remarks>
+    public interface IBioPolymerWithSetMods : IHasChemicalFormula
+    {
+        string BaseSequence { get; }
+        string FullSequence { get; }
+        double MostAbundantMonoisotopicMass { get; }
+        string SequenceWithChemicalFormulas { get; }
+        int OneBasedStartResidue { get; }
+        int OneBasedEndResidue { get; }
+        int MissedCleavages { get; }
+        CleavageSpecificity CleavageSpecificityForFdrCategory { get; set; }
+        char PreviousResidue { get; }
+        char NextResidue { get; }
+        IDigestionParams DigestionParams { get; }
+        Dictionary<int, Modification> AllModsOneIsNterminus { get; }
+        int NumMods { get; }
+        int NumFixedMods { get; }
+        int NumVariableMods { get; }
+        int Length { get; }
+        char this[int zeroBasedIndex] => BaseSequence[zeroBasedIndex];
```

Comment:
am i confused? one is n terminus? not zero based index.

Reply:
This is how the sequence is indexed for digestion. The modifications are indexed based upon one being N-terminal. This is maintained from previous implementations